File: apps/demo-be-csas/tests/DemoBeCsas.Tests/Integration/Steps/HealthSteps.cs
using System.Text.Json;
using DemoBeCsas.Tests.ScenarioContext;
using FluentAssertions;
using Reqnroll;
using Xunit;

namespace DemoBeCsas.Tests.Integration.Steps;

[Binding]
[Trait("Category", "Integration")]
public class HealthSteps(TestWebApplicationFactory factory, SharedState state)
{
    [When(@"^an operations engineer sends GET \/health$")]
    public async Task WhenGetHealth()
    {
        var client = factory.CreateClient();
        state.LastResponse = await client.GetAsync("/health");
    }

    [When(@"^an unauthenticated engineer sends GET \/health$")]
    public async Task WhenUnauthenticatedGetHealth()
    {
        var client = factory.CreateClient();
        state.LastResponse = await client.GetAsync("/health");
    }

    [Then("the health status should be {string}")]
    public async Task ThenHealthStatus(string expectedStatus)
    {
        state.LastResponse.Should().NotBeNull();
        var body = await state.LastResponse!.Content.ReadAsStringAsync();
        using var doc = JsonDocument.Parse(body);
        doc.RootElement.GetProperty("status").GetString().Should().Be(expectedStatus);
    }

    [Then("the response should not include detailed component health information")]
    public async Task ThenNoDetailedHealthInfo()
    {
        state.LastResponse.Should().NotBeNull();
        var body = await state.LastResponse!.Content.ReadAsStringAsync();
        body.Should().NotContain("components");
        body.Should().NotContain("details");
    }
}

File: apps/demo-be-python-fastapi/tests/integration/steps/health_steps.py
"""BDD step definitions for health check feature."""

from pytest_bdd import scenarios, then, when

from tests.integration.conftest import GHERKIN_ROOT
from tests.integration.service_client import FakeResponse, ServiceClient

scenarios(str(GHERKIN_ROOT / "health" / "health-check.feature"))


@when("an operations engineer sends GET /health", target_fixture="response")
def get_health(client: ServiceClient) -> FakeResponse:
    return client.get_health()


@when("an unauthenticated engineer sends GET /health", target_fixture="response")
def get_health_unauthenticated(client: ServiceClient) -> FakeResponse:
    return client.get_health()


@then('the health status should be "UP"')
def check_health_status(response: FakeResponse) -> None:
    body = response.json()
    assert body.get("status") == "UP", f"Expected status=UP, got: {body}"


@then("the response should not include detailed component health information")
def check_no_component_details(response: FakeResponse) -> None:
    body = response.json()
    assert "components" not in body
    assert "details" not in body
    assert set(body.keys()) == {"status"}, f"Unexpected keys: {set(body.keys())}"
